client/layout/masterbar/logged-in.jsx

```diff
@@ -234,7 +235,7 @@ class MasterbarLoggedIn extends Component {
 			? domainManagementList( siteSlug, currentRoute, true )
 			: '/sites';
 		if ( hostingDashboardOptIn ) {
-			mySitesUrl = domainOnlySite ? '/v2/domains' : '/v2/sites';
+			mySitesUrl = domainOnlySite ? dashboardLink( '/domains' ) : dashboardLink( '/sites' );
```

Comment:
This is for the masterbar:

<img width="1433" height="394" alt="Image" src="https://github.com/user-attachments/assets/a8c9e141-4382-47c2-a511-857710feb3dc" />